### PR TITLE
WIP schema for video transcripts

### PR DESCRIPTION
**Pull request by horlami228**
### Description

- Added MongoDB schemas for `Video`, `VideoParticipant`, `VideoDialogue`, and `VideoTranscript`.
- Integrated Zod for schema validation to ensure data integrity.

### CODE REVIEW

File: schema/mongoose.ts [hunk 1]
```diff
@@ -789,6 +789,75 @@ export const Video = new Schema<types.Video>(
...
+  }
+)
+const VideoDialogueSchema = new mongoose.Schema({
+  participantId: { type: mongoose.Schema.Types.ObjectId, ref: 'VideoParticipant', required: true },
+  text: { type: String, required: true },
+  timestamp: { type: String, required: true },
+}, {
+  collection: 'VideoDialogue',
+  timestamps: { createdAt: 'createdDate', updatedAt: 'updatedDate' },
+});
```
Comment (hashwarp) [inline schema/mongoose.ts]: looks like old version

File: schema/mongoose.ts [hunk 2]
```diff
@@ -789,6 +789,75 @@ export const Video = new Schema<types.Video>(
...
 );
 
+export const VideoParticipant = new Schema<types.VideoParticipant>(
+  {
+    name: { type: String, required: true },
+    profileId:  { type: mongoose.Schema.Types.ObjectId, ref: 'Profile', required: true },
+    description: { type: String },
+    status: { type: String, default: 'Active', enum: ['Paused', 'Pending', 'Active', 'Archived'] },
+    data: { type: Object, default: {} },
+    meta: { type: Object, default: {} },
```
Comment (hashwarp) [inline schema/mongoose.ts:806]: good, you noticed every schema needs a meta (and many have data). they also all need applicationId:


    applicationId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Application',
      required: true,
    },

File: schema/mongoose.ts [hunk 3]
```diff
@@ -789,6 +789,75 @@ export const Video = new Schema<types.Video>(
   { collection: 'Video', timestamps: { createdAt: 'createdDate', updatedAt: 'updatedDate' } }
 );
 
+export const VideoParticipant = new Schema<types.VideoParticipant>(
+  {
+    name: { type: String, required: true },
+    profileId:  { type: mongoose.Schema.Types.ObjectId, ref: 'Profile', required: true },
```
Comment (hashwarp) [inline schema/mongoose.ts]: not required, because we wont always have a profile for every participant in the system

File: schema/mongoose.ts [hunk 4]
```diff
@@ -789,6 +789,75 @@ export const Video = new Schema<types.Video>(
...
+  text: { type: String, required: true },
+  timestamp: { type: String, required: true },
+}, {
+  collection: 'VideoDialogue',
+  timestamps: { createdAt: 'createdDate', updatedAt: 'updatedDate' },
+});
+
+export const VideoDialogue = new Schema<types.VideoDialogue>(
+  {
+    participantId: { type: mongoose.Schema.Types.ObjectId, ref: 'VideoParticipant', required: true },
```
Comment (hashwarp) [inline schema/mongoose.ts]: probably dont wanna require this either

File: schema/mongoose.ts [hunk 5]
```diff
@@ -789,6 +789,75 @@ export const Video = new Schema<types.Video>(
...
+    name: { type: String, required: true },
```
Comment (hashwarp) [inline schema/mongoose.ts:825]: is name/description needed?

File: schema/mongoose.ts (hunk 2 above)
Comment (horlami228) [reply schema/mongoose.ts:806]: i would include it from now on

File: schema/mongoose.ts (hunk 3 above)
Comment (horlami228) [reply schema/mongoose.ts]: okay would change the contrainst on that

File: schema/mongoose.ts (hunk 4 above)
Comment (horlami228) [reply schema/mongoose.ts]: alright boss

File: schema/mongoose.ts (hunk 5 above)
Comment (horlami228) [reply schema/mongoose.ts:825]: will check to see if its important there

**Pull request merged**